metadata-ingestion/examples/library/dataset_query_deprecation.py

```diff
@@ -1,21 +1,19 @@
-from datahub.emitter.mce_builder import make_dataset_urn
+from typing import Optional
 
-# read-modify-write requires access to the DataHubGraph (RestEmitter is not enough)
-from datahub.ingestion.graph.client import DatahubClientConfig, DataHubGraph
-
-# Imports for metadata model classes
 from datahub.metadata.schema_classes import DeprecationClass
+from datahub.sdk import DataHubClient, DatasetUrn
 
-dataset_urn = make_dataset_urn(platform="hive", name="fct_users_created", env="PROD")
-
-gms_endpoint = "http://localhost:8080"
-graph = DataHubGraph(DatahubClientConfig(server=gms_endpoint))
+client = DataHubClient.from_env()
 
-# Query multiple aspects from entity
-result = graph.get_aspects_for_entity(
-    entity_urn=dataset_urn,
-    aspects=["deprecation"],
-    aspect_types=[DeprecationClass],
+dataset = client.entities.get(
+    DatasetUrn(platform="hive", name="fct_users_created", env="PROD")
 )
 
-print(result)
+# Check if dataset is deprecated
+deprecation: Optional[DeprecationClass] = dataset._get_aspect(DeprecationClass)
+if deprecation and deprecation.deprecated:
+    print(f"Dataset is deprecated: {deprecation.note}")
+    if hasattr(deprecation, "decommissionTime"):
```

Comment:
why the `hasattr`?

considering

```
class DeprecationClass(_Aspect):
    """Deprecation status of an entity"""


    ASPECT_NAME = 'deprecation'
    ASPECT_INFO = {}
    RECORD_SCHEMA = get_schema_type("com.linkedin.pegasus2avro.common.Deprecation")

    def __init__(self,
        deprecated: bool,
        note: str,
        actor: str,
        decommissionTime: Union[None, int]=None,
        replacement: Union[None, str]=None,
    ):
```

we could

```suggestion
    if deprecation.decommissionTime:
```

Reply:
fixed